### ISSUE TITLE
Improve Symja Project structure

### ISSUE THREAD

**Issue by HannesWell** [axkr/symja_android_library#271]
I would like to suggest and discuss the following improvements for the project structure of Symja.

- Move the `matheclipse` or `symja_android_library` module to the top of the repository
  - It reduces nesting
  - This simplifies the import of the Symja project into an IDE workspace (at least when using Eclipse, because you can simply import everything)
  - Most Maven projects I know have their root Maven module in their repository root directory.

- Rename `fronted` module to `parser`
- Split the embedded dependencies into own modules as suggested in #259
- Move more advanced 'UI'-like IO/import/export functionality from core to another module. Maybe a new module, maybe into the IO module.
  - the goal is to further reduce the dependencies of the core module to e.g. jgrapht.io, antlr.
  I have to figure out in detail if it is possible.

- Clean up 
  - make tools/toolslib a maven module and merge them if possible, so they can be used more easily. For what 
  - Merge .m files into one folder
    - A.t.m. .m files are at different locations `VectorAnalysis.m`, `Rubi` folder, `rules` folder
  - Check which files in repo root can be moved to a more appropriated location or can be removed
    - E.g. is `bitbucket-pipelines.yml` used?
    - the files `en_badge_web_generic.png` `console.gif` could be moved to screenshots
    - This could be done similarly for the symja_android_library sub-folder

When moving files around, care should be taken to update links to that files accordingly.

**Comment by axkr** [comment]
> I would like to suggest and discuss the following improvements for the project structure of Symja.
> 
>     * Move the `matheclipse` or `symja_android_library` module to the top of the repository

-1 from my side, because every link in the wiki and in the `symja_android_library/doc` folder must be reworked. I think we should first get a release 2.0 out, before restructuring everything.


>     * Rename `fronted` module to `parser`

+1 we can of course rename it. 

>     * Split the embedded dependencies into own modules as suggested in [Split embedded dependencies up into own modules #259](https://github.com/axkr/symja_android_library/issues/259)

+ 1 this seems to be important to get a release 2.0 for Maven Central.

>     * Move more advanced 'UI'-like IO/import/export functionality from core to another module. Maybe a new module, maybe into the IO module.
>       
>       * the goal is to further reduce the dependencies of the core module to e.g. jgrapht.io, antlr.
>         I have to figure out in detail if it is possible.
-1 for jgrapht we will have to use anyway it in the future for more adavanced algorithms 
+1 for antlr if it's possible. But I once requested it from LogicNG: https://github.com/logic-ng/LogicNG/issues/14 
So for a first 2.0 release we should include antlr runtime.

>     * Clean up
>       
>       * make tools/toolslib a maven module and merge them if possible, so they can be used more easily. 

+1 lgtm

>       * For what Merge .m files into one folder
>         
>         * A.t.m. .m files are at different locations `VectorAnalysis.m`, `Rubi` folder, `rules` folder
>       * Check which files in repo root can be moved to a more appropriated location or can be removed
>         
>         * E.g. is `bitbucket-pipelines.yml` used?

+1 as we now have github actions we can switch of bitbucket pipelines: https://bitbucket.org/axelclk/symja_android_library/addon/pipelines/home


>         * the files `en_badge_web_generic.png` `console.gif` could be moved to screenshots
+1 

>         * This could be done similarly for the symja_android_library sub-folder
-1 as written above, we should first get a release 2.0 out, before restructuring everything.

**Comment by HannesWell** [comment]
Thanks for your remarks.

> -1 from my side, because every link in the wiki and in the `symja_android_library/doc` folder must be reworked. I think we should first get a release 2.0 out, before restructuring everything.

Sure, most of the refactoring suggested is not urgent and can be postponed after the 2.0.0 release without any problem.
The only thing that should be done before are the changes that affect Symja's Maven modules and their names, but since you agree with them, that's fine. 👍🏽

**Comment by HannesWell** [comment]
To update the links in the wiki one can simply checkout the wiki as git repo, simply search-and-replace all link roots and commit them:
https://stackoverflow.com/questions/1170166/can-i-checkout-github-wikis-like-a-git-repository
